src/test/java/com/ksm/wstbackoffice/controller/AddressControllerTest.java

```diff
@@ -0,0 +1,80 @@
+package com.ksm.wstbackoffice.controller;
+
+import io.restassured.RestAssured;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.hamcrest.Matchers.is;
+
+@SpringBootTest
+@ActiveProfiles("test")
+public class AddressControllerTest {
+    @Value("${server.port}")
+    private int port;
+
+    @Test
+    public void findByIdTest() {
+        given()
+                .baseUri(RestAssured.DEFAULT_URI)
+                .port(port)
+                .basePath("/addresses/{id}")
+                .pathParam("id", 1)
+                .when()
+                .get()
+                .then()
+                .body(
+                        "id", is(1)
+                )
+                .statusCode(200);
+    }
+
+    @Test
+    public void findAllTest() {
+        given()
+                .baseUri(RestAssured.DEFAULT_URI)
+                .port(port)
+                .basePath("/addresses")
+                .when()
+                .get()
+                .then()
+                .body("isEmpty()", Matchers.is(false))
```

Comment:
Here you are verifying that body is not empty. Where do you populate the data to be sure it is not empty

Reply:
changed
Assert.assertNotNull(addresses);
        assertThat(addresses.size(), Matchers.greaterThanOrEqualTo(0));

Reply:
Where do you populate the data to be sure it is not empty before running this test

Reply:
done